sched/queue/memory/simple.go

```diff
@@ -0,0 +1,81 @@
+package memory
+
+// In-memory Scheduler Queue
+
+import (
+	"fmt"
+	"github.com/scootdev/scoot/sched"
+	"github.com/scootdev/scoot/sched/queue"
+	"time"
+)
+
+func NewSimpleQueue() (queue.Queue, chan sched.Job) {
+	q := &simpleQueue{}
+	q.outCh = make(chan sched.Job, 1)
+	q.inCh = make(chan enqueueReq)
+	go q.loop()
+	return q, q.outCh
+}
+
+type simpleQueue struct {
+	nextID int
+	inCh   chan enqueueReq
+	outCh  chan sched.Job
+}
+
+type enqueueReq struct {
+	job      sched.Job
+	resultCh chan enqueueResult
+}
+
+type enqueueResult struct {
+	jobID string
+	err   error
+}
+
+func (q *simpleQueue) Enqueue(job sched.Job) (string, error) {
+	if job.Id != "" {
+		return "", queue.NewInvalidJobRequest(fmt.Sprintf("job.Id must be unset; was %v", job.Id))
+	}
+	if len(job.Tasks) == 0 {
+		return "", queue.NewInvalidJobRequest("job must have at least 1 task")
+	}
+	for _, task := range job.Tasks {
+		if task.Id == "" {
```

Comment:
Encapsulate in Queue package. (Or should this even just be moved to the sched package, where Job is defined?)